mesonbuild/environment.py

```diff
@@ -1063,7 +1063,7 @@ def sanitize(p):
                     m = 'Failed to detect MSVC compiler version: stderr was\n{!r}'
                     raise EnvironmentException(m.format(err))
                 cl_signature = lookat.split('\n')[0]
-                match = re.search('.*(x86|x64|ARM|ARM64)$', cl_signature)
+                match = re.search('.*(x86|x64|ARM|ARM64)( |$)', cl_signature)
```

Comment:
If we set `VSLANG=1033` in the env, that will fix this issue too. It will force MSVC to use the eng codepage and eliminate locale-specific issues in general.

Reply:
The problem is that VSLANG is not always honoured by cl.exe. See https://github.com/mesonbuild/meson/issues/6128#issuecomment-551734903 or https://bugreports.qt.io/plugins/servlet/mobile#issue/QTCREATORBUG-20327/comment/400818

Reply:
Ah, interesting, thanks :)